### PR TITLE
Enable Werror for inference tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Disable warnings suppression
 - Fix warnings

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/inference/dev_api/memory_solver.hpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ class MemorySolver {
 
         // remove unused timestamps (not a begin of some box)
         // each ts should start a box
-        std::vector<bool> ts_exist(max_ts + 1);
+        std::vector<bool> ts_exist((size_t)max_ts + 1);
```
Comment (ilyachur) [inline src/inference/dev_api/memory_solver.hpp:84]: What it `max_ts` < 0?

File: src/inference/tests/functional/blob_copy_test.cpp [hunk 2]
```diff
@@ -80,7 +80,7 @@ InferenceEngine::Blob::Ptr createBlob(InferenceEngine::Precision precision,
 size_t GenerateRandom(size_t elem) {
     size_t result;
     do {
-        result = std::floor(std::rand() / static_cast<float>(RAND_MAX * elem));
+        result = (size_t)std::floor(std::rand() / static_cast<float>(RAND_MAX * elem));
```
Comment (ilyachur) [inline src/inference/tests/functional/blob_copy_test.cpp:83]: Please use static cast instead of C style cast

File: src/inference/dev_api/memory_solver.hpp (hunk 1 above)
Comment (olpipi) [reply src/inference/dev_api/memory_solver.hpp]: Idk why I added these changed to api.
Removed

File: src/inference/tests/functional/blob_copy_test.cpp (hunk 2 above)
Comment (olpipi) [reply src/inference/tests/functional/blob_copy_test.cpp]: done

File: src/inference/tests/unit/ie_extension_test.cpp [hunk 3]
```diff
@@ -17,6 +17,8 @@ using namespace InferenceEngine;
 
 using ExtensionTests = ::testing::Test;
 
+std::string getExtensionPath();
```
Comment (ilyachur) [inline src/inference/tests/unit/ie_extension_test.cpp:20]: Why do you need this forward declaration?

File: src/inference/tests/functional/caching_test.cpp [hunk 4]
```diff
@@ -54,6 +54,8 @@ static const std::vector<std::string> cacheFolders{
     std::string("testCache"),
 };
 
+std::string getTestCaseName(const testing::TestParamInfo<std::tuple<TestParam, std::string>>& obj);
```
Comment (ilyachur) [inline src/inference/tests/functional/caching_test.cpp:57]: Why did you add declaration before definition?
Looks strange, I don't believe that compiler reports warning for this.

File: src/inference/tests/functional/caching_test.cpp [hunk 5]
```diff
@@ -2035,7 +2037,7 @@ TEST_P(CachingTest, LoadAUTO_OneDevice) {
     });
     std::string cacheDir = m_cacheDir;
     MkDirGuard guard(cacheDir);
-    for (index; index < TEST_COUNT; index++) {
+    for (; index < TEST_COUNT; index++) {
```
Comment (ilyachur) [inline src/inference/tests/functional/caching_test.cpp:2040]: I think we can remove local variable and move it to the cycle.

File: src/inference/tests/unit/ie_extension_test.cpp (hunk 3 above)
Comment (ilyachur) [reply src/inference/tests/unit/ie_extension_test.cpp:20]: The same comment as before....
I don't believe that we need declaration before definition

File: src/inference/tests/functional/caching_test.cpp (hunk 4 above)
Comment (alalek) [reply src/inference/tests/functional/caching_test.cpp:57]: actually compiler reports...

'static' should be used to mark function as local instead of forward declaration.

Comment (olpipi) [reply src/inference/tests/functional/caching_test.cpp]: @alalek  thx for advice

@ilyachur fixed

File: src/inference/tests/functional/caching_test.cpp (hunk 5 above)
Comment (olpipi) [reply src/inference/tests/functional/caching_test.cpp]: fixed

**Review by ilyachur — APPROVED** [review]
